test/Microsoft.AspNetCore.Mvc.Core.Test/ContentResultTest.cs

```diff
@@ -167,6 +142,142 @@ public class ContentResultTest
             var finalResponseContentType = httpContext.Response.ContentType;
             Assert.Equal(expectedContentType, finalResponseContentType);
             Assert.Equal(expectedContentData, memoryStream.ToArray());
+            Assert.Equal(expectedContentData.Length, httpContext.Response.ContentLength);
+        }
+
+        public static TheoryData<string, string> ContentResult_WritesDataCorrectly_ForDifferentContentSizesData
```

Comment:
@rynowak I feel these tests need to be ideally in HttpAbstractions(I checked the tests there and I think they are missing some coverage (ex: surrogate pair tests)) as they are more like testing the response stream writer. We could keep few sanity level tests here, thoughts?